### PR TITLE
DevWs for Smartenit, Inc containing containing Testing

### PR DESCRIPTION
**Pull request by lecontr**

### CODE REVIEW

File: devicetypes/smartenit/iot8-z.src/iot8-z.groovy [hunk 1]
```diff
@@ -0,0 +1,238 @@
...
+        capability "Refresh"
+        capability "Health Check"
+        capability "Sensor"
+        capability "Switch"
+        capability "monthpublic25501.analogSensor"
+        
+        command "childOn", ["string"]
+        command "childOff", ["string"]
+        
+        fingerprint manufacturer: "Smartenit, Inc", model: "IOT8-Z", deviceJoinName: "SMRT-IOT ", profileId: "0104", inClusters: "0000, 0003, 0006, 000C, 000F", outClusters: "0019"
```
Comment (PKacprowiczS) [inline devicetypes/smartenit/iot8-z.src/iot8-z.groovy]: I'm assuming this is switch device, so change `deviceJoinName` to `"Smartenit Switch"`

File: devicetypes/smartenit/iot8-z.src/iot8-z.groovy [hunk 2]
```diff
@@ -0,0 +1,238 @@
...
+        capability "monthpublic25501.analogSensor"
+        
+        command "childOn", ["string"]
+        command "childOff", ["string"]
+        
+        fingerprint manufacturer: "Smartenit, Inc", model: "IOT8-Z", deviceJoinName: "SMRT-IOT ", profileId: "0104", inClusters: "0000, 0003, 0006, 000C, 000F", outClusters: "0019" 
+	}
+
+
+	simulator {
```
Comment (PKacprowiczS) [inline devicetypes/smartenit/iot8-z.src/iot8-z.groovy]: Remove `simulator {}` and `tiles {}` blocks.

File: devicetypes/smartenit/iot8-z.src/iot8-z.groovy [hunk 3]
```diff
@@ -0,0 +1,238 @@
...
+
+def refresh() {
+	def refreshCommands = zigbee.onOffRefresh() 
+    def numberOfChildDevices = 8
+    
+	for(def endpoint : 2..numberOfChildDevices) {
+		refreshCommands += zigbee.readAttribute(zigbee.ONOFF_CLUSTER, 0x0000, [destEndpoint: endpoint])
+	}
+    for(def endpoint : 1..4) {
+    	refreshCommands += zigbee.readAttribute(0x000F, 0x0055, [destEndpoint: endpoint])
```
Comment (PKacprowiczS) [inline devicetypes/smartenit/iot8-z.src/iot8-z.groovy]: Please change all "magic numbers" to constants, like:
`private getBATTERY_PERCENTAGE_REMAINING() { 0x0021 }`
to which later you can refer as
`BATTERY_PERCENTAGE_REMAINING`
(this particular example is from https://github.com/SmartThingsCommunity/SmartThingsPublic/blob/master/devicetypes/smartthings/leaksmart-water-sensor.src/leaksmart-water-sensor.groovy#L62 )

**Review by PKacprowiczS — CHANGES_REQUESTED** [review]
Please be consistent with your code, e.g. use tabs as indentation. Generally speaking follow guidelines from https://github.com/greens/DTH_development

**Comment by lecontr** [comment]
Okay excellent. Thank you so much for the feedback! I'm currently working on fixing those issues. 🙂 

I was wondering about one thing though. 😮  I have some custom child device handlers that I created for this project. How can I submit those?

**Comment by PKacprowiczS** [comment]
Can't you just include them in this PR?

**Comment by lecontr** [comment]
Ah thank you. I've made the alterations. 🙂

File: devicetypes/smartenit/iot8-z.src/iot8-z.groovy [hunk 4]
```diff
@@ -0,0 +1,223 @@
...
+		capability "Refresh"
+		capability "Health Check"
+		capability "Sensor"
+		capability "Switch"
+		capability "monthpublic25501.analogSensor"
+
+		command "childOn", ["string"]
+		command "childOff", ["string"]
+
+		fingerprint manufacturer: "Smartenit, Inc", model: "IOT8-Z", deviceJoinName: "Smartenit Switch Sensor ", profileId: "0104", inClusters: "0000, 0003, 0006, 000C, 000F", outClusters: "0019"
```
Comment (PKacprowiczS) [inline devicetypes/smartenit/iot8-z.src/iot8-z.groovy]: Please change `deviceJoinName` to `Smartenit Switch`.

File: devicetypes/smartenit/iot8-z.src/iot8-z.groovy [hunk 5]
```diff
@@ -0,0 +1,223 @@
...
+			log.debug "Exception creating child device: ${e}"
+		}
+	}
+}
+
+def configure() {
+	log.debug "configure"
+    
+	configureHealthCheck()
+	def configurationCommands = zigbee.configureReporting(getBINARY_INPUT_CLUETER(), getPRESENT_VALUE_ATTRIBUTE(), 0x10, 10, 600, null)
```
Comment (PKacprowiczS) [inline devicetypes/smartenit/iot8-z.src/iot8-z.groovy]: Side note, if you define get method like you did
```
private getANALOG_INPUT_CLUSTER() { 0x000C }
```
you can access value by simple `ANALOG_INPUT_CLUSTER`.

File: devicetypes/smartenit/iot8-z.src/iot8-z.groovy [hunk 6]
```diff
@@ -0,0 +1,223 @@
...
+                
+				int intValue = (int) ceilingVal
+				return createEvent(name: "inputValue", value: intValue)
+			}
+		} else {
+			def childDevice = childDevices.find {
+				it.deviceNetworkId == "$device.deviceNetworkId:${eventDescMap.sourceEndpoint}" || it.deviceNetworkId == "$device.deviceNetworkId:${eventDescMap.endpoint}"
+			}
+			if (childDevice) {
+				if (eventDescMap?.clusterId == "000F" || eventDescMap?.cluster == "000F") {
```
Comment (PKacprowiczS) [inline devicetypes/smartenit/iot8-z.src/iot8-z.groovy]: Each of `eventDescMap.clusterId` and `eventDescMap.cluster` have specific type, first is integer, while other is string, so one of the conditions will always be false.  
Furthermore, you can use integer value to compare it to constants you've defined above.

File: devicetypes/smartenit/iot8-z-child-analog-contact-switch.src/iot8-z-child-analog-contact-switch.groovy [hunk 7]
```diff
@@ -0,0 +1,44 @@
...
+	log.debug "Installed"
+}
+
+def updated() {
+	log.debug "Updated"
+}
+
+
+def parse(String description) {
+}
```
Comment (PKacprowiczS) [inline devicetypes/smartenit/iot8-z-child-analog-contact-switch.src/iot8-z-child-analog-contact-switch.groovy]: `installed()`, `updated()` and `parse()` methods are not necessary in child DTHes.

**Review by PKacprowiczS — CHANGES_REQUESTED** [review]

**Comment by k-morawski** [comment]
Dear @lecontr , could you please update your code according to @PKacprowiczS feedback? Many thanks!

**Comment by lecontr** [comment]
Certainly! Apologies for the delay.

File: devicetypes/smartenit/iot8-z.src/iot8-z.groovy (hunk 6 above)
Comment (lecontr) [reply devicetypes/smartenit/iot8-z.src/iot8-z.groovy]: I was a bit puzzled by this feedback. I believe that the `eventDescMap.clusterId` type in this case is a string based on the function call I'm using, which is `zigbee.parseDescriptionAsMap`. I took a look at other examples and found that some implementations use `zigbee.parse` which result in an integer type. 🙂

Comment (PKacprowiczS) [reply devicetypes/smartenit/iot8-z.src/iot8-z.groovy]: Sorry for misleading you, `clusterInt` was the one I was looking for.
Example of usage: https://github.com/SmartThingsCommunity/SmartThingsPublic/blob/a7c980340992ca37508d42a70f354b5fd7d30048/devicetypes/smartthings/zigbee-power-meter.src/zigbee-power-meter.groovy#L58

Comment (lecontr) [reply devicetypes/smartenit/iot8-z.src/iot8-z.groovy]: Ah ok. I understand now. 🙂 Thank you.

File: devicetypes/smartenit/iot8-z.src/iot8-z.groovy [hunk 8]
```diff
@@ -0,0 +1,223 @@
...
+}
+
+// parse events into attributes
+def parse(String description) {	
+	Map eventMap = zigbee.getEvent(description)
+	Map eventDescMap = zigbee.parseDescriptionAsMap(description)
+
+	if (eventMap) {
+		if ((eventDescMap?.sourceEndpoint == "01") || (eventDescMap?.endpoint == "01")) {
+			if (eventDescMap?.clusterId == "000C") {
```
Comment (PKacprowiczS) [inline devicetypes/smartenit/iot8-z.src/iot8-z.groovy]: You've missed this one.

File: devicetypes/smartenit/iot8-z.src/iot8-z.groovy [hunk 9]
```diff
@@ -0,0 +1,223 @@
...
+	log.debug "configure"
+    
+	configureHealthCheck()
+	def configurationCommands = zigbee.configureReporting(BINARY_INPUT_CLUSTER, PRESENT_VALUE_ATTRIBUTE, 0x10, 10, 600, null)
+	for (def endpoint: 2..4) {
+		configurationCommands += zigbee.configureReporting(BINARY_INPUT_CLUSTER, PRESENT_VALUE_ATTRIBUTE, 0x10, 10, 600, null, [destEndpoint: endpoint])
+	}
+	
+	configurationCommands += zigbee.onOffConfig(0, 120)
+	for(def endpoint : 2..8) {
```
Comment (PKacprowiczS) [inline devicetypes/smartenit/iot8-z.src/iot8-z.groovy]: nit: missing space

File: devicetypes/smartenit/iot8-z.src/iot8-z.groovy [hunk 10]
```diff
@@ -0,0 +1,223 @@
...
+
+def ping() {
+	refresh()
+}
+
+def refresh() {
+	def refreshCommands = zigbee.onOffRefresh() 
+	def numberOfChildDevices = 8
+
+	for(def endpoint : 2..numberOfChildDevices) {
```
Comment (PKacprowiczS) [inline devicetypes/smartenit/iot8-z.src/iot8-z.groovy]: nit: missing space

File: devicetypes/smartenit/iot8-z.src/iot8-z.groovy [hunk 11]
```diff
@@ -0,0 +1,223 @@
...
+		refreshCommands += zigbee.readAttribute(zigbee.ONOFF_CLUSTER, ONOFF_ATTRIBUTE, [destEndpoint: endpoint])
+	}
+	for(def endpoint : 1..4) {
```
Comment (PKacprowiczS) [inline devicetypes/smartenit/iot8-z.src/iot8-z.groovy]: nit: missing space

File: devicetypes/smartenit/iot8-z.src/iot8-z.groovy [hunk 12]
```diff
@@ -0,0 +1,223 @@
...
+		capability "Refresh"
+		capability "Health Check"
+		capability "Sensor"
+		capability "Switch"
+		capability "monthpublic25501.analogSensor"
+
+		command "childOn", ["string"]
+		command "childOff", ["string"]
+
+		fingerprint manufacturer: "Smartenit, Inc", model: "IOT8-Z", deviceJoinName: "Smartenit Switch ", profileId: "0104", inClusters: "0000, 0003, 0006, 000C, 000F", outClusters: "0019"
```
Comment (PKacprowiczS) [inline devicetypes/smartenit/iot8-z.src/iot8-z.groovy]: unnecessary space at the end of `deviceJoinName: "Smartenit Switch "`

**Review by PKacprowiczS — CHANGES_REQUESTED** [review]

**Comment by k-morawski** [comment]
Dear @lecontr , @PKacprowiczS is on leave today but will check the changes on Monday next week. Best regards

**Comment by lecontr** [comment]
Oh no problem. 🙂 Thank you for the heads up.

**Review by PKacprowiczS — DISMISSED** [review]
@greens

File: devicetypes/smartenit/iot8-z.src/iot8-z.groovy [hunk 13]
```diff
@@ -0,0 +1,223 @@
...
+					isComponent: false
+					])
+			} else if (endpoint >= 3 && endpoint <= 4) {
+				addChildDevice("Smartenit", "IOT8-Z-child-contact-switch", "${device.deviceNetworkId}:0${endpoint}", device.hubId,
+					[completedSetup: true,
+					label: "${device.displayName} ${endpoint}",
+					isComponent: false
+					])
+			} else if (endpoint >= 5 && endpoint <= 8) {
+				addChildDevice("smartthings", "ZigBee Switch", "${device.deviceNetworkId}:0${endpoint}", device.hubId,
```
Comment (greens) [inline devicetypes/smartenit/iot8-z.src/iot8-z.groovy]: I wouldn't expect this to work well. If the user tries to turn these switches on and off, I expect the command will fail. Perhaps you meant to use `"Child Switch"`?

Comment (lecontr) [reply devicetypes/smartenit/iot8-z.src/iot8-z.groovy]: Oh you're right. This was leftover from my testing and it's been working for me so I completely missed it. Let me make the modification real quick.

**Review by greens — APPROVED** [review]

**Comment by greens** [comment]
Will be live on Production on the 27th

**Pull request merged**